frontend/src/components/CompareTable.test.tsx

```diff
@@ -52,8 +56,34 @@ describe('CompareTable', () => {
     );
   });
 
+  it('logs error if xlabels and xparentlabels have different lengths', () => {
+    shallow(
```

Comment:
Although existing tests are using snapshot, we can still introduce new testing using react-testing-library by creating a different `describe()` in the same file. The existing tests won't be affected in this case.

Reply:
This makes sense - I've introduced a separate `describe` function for the new tests dealing specifically with `xParentLabels`: https://github.com/kubeflow/pipelines/pull/8024/commits/2d4e6057875c30c5d7e55bf06a563b37ff256733